pool.go

```diff
@@ -166,6 +178,70 @@ func (p *pool) refreshPool() {
 	}
 }
 
+func (p *pool) checkPool() {
```

Comment:
Please can we add a unit test for check pool for block reqs and car reqs and for main nodes vs unknown nodes ?

Reply:
I meant if the node is already an unknown node, don't do mirroring vs do mirroring if the node is a main node.